In [1]:
library(faraway)
library(car)
library(MASS)
library(lmtest)
library(tidyverse)
library(nlme)

Loading required package: carData


Attaching package: ‘car’


The following objects are masked from ‘package:faraway’:

    logit, vif


Loading required package: zoo


Attaching package: ‘zoo’


The following objects are masked from ‘package:base’:

    as.Date, as.Date.numeric


── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
✖ dplyr::recode() masks car::recode()
✖ dplyr::select() masks MASS::select()
✖ purrr::some()   masks car::some()


Attaching package: ‘nlme’


The following object is masked from ‘package:dplyr’:

    collapse




In [2]:
df_software <- read_table("software_sales.txt") 
df_software$Software <- as.factor(df_software$Software)


── Column specification ────────────────────────────────────────────────────────
cols(
  Rep = col_double(),
  Software = col_double(),
  SalesLastQuarter = col_double(),
  SalesThisQuarter = col_double()
)



In [3]:
#5A
sw_model1 <- lm(SalesThisQuarter ~ . - SalesLastQuarter - Rep, data = df_software)
summary(sw_model1)


Call:
lm(formula = SalesThisQuarter ~ . - SalesLastQuarter - Rep, data = df_software)

Residuals:
    Min      1Q  Median      3Q     Max 
-19.583  -6.833   1.417   7.583  32.417 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   81.583      3.281  24.866   <2e-16 ***
Software2     -2.000      4.640  -0.431    0.669    
Software3     -7.667      4.640  -1.652    0.108    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 11.37 on 33 degrees of freedom
Multiple R-squared:  0.08176,	Adjusted R-squared:  0.02611 
F-statistic: 1.469 on 2 and 33 DF,  p-value: 0.2448


y = 81.6 - 2.00 $x_1$ - 7.67 $x_2$   
where:    
$x_1$ -> Whether Software 2 was used  
$x_2$ -> Whether Software 3 was used 

Part i)

$E(y | x_1 = 1, x_2 = 0) = (81.6 - 2.00) - 7.67 * 0 = 79.42 $   
$E(y | x_1 = 0, x_2 = 1) = (81.6 - 7.67) - 2.00 * 0 = 73.93 $ 

In [4]:
# Part ii)
anova(sw_model1)

sprintf("Variance explained by the software is %0.3f", (anova(sw_model1)[1,2] / sum(anova(sw_model1)[,2])))

Df Sum Sq    Mean Sq  F value  Pr(>F)   
Software   2  379.5556 189.7778 1.469161 0.2447802
Residuals 33 4262.7500 129.1742       NA        NA

[1] "Variance explained by the software is 0.082"

*Part iii)*

$H_0$ states that software has no effect on sales.  
$H_A$ states that software does have an effect on sales.

The *f statistic* given from the anova table, 1.47, and the corresponding *p value*, 0.25, indicate that we do not have enough evidence to reject the null hypothesis at the 0.05 significance level.

In [5]:
#5B
sw_model2 <- lm(SalesThisQuarter ~ . -Rep, data = df_software)
summary(sw_model2)


Call:
lm(formula = SalesThisQuarter ~ . - Rep, data = df_software)

Residuals:
    Min      1Q  Median      3Q     Max 
-18.974  -4.194   0.554   3.536  16.049 

Coefficients:
                 Estimate Std. Error t value Pr(>|t|)    
(Intercept)      -36.4423    16.4223  -2.219   0.0337 *  
Software2          0.7535     2.9249   0.258   0.7984    
Software3         -1.2835     3.0311  -0.423   0.6748    
SalesLastQuarter   1.5019     0.2073   7.244 3.14e-08 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 7.104 on 32 degrees of freedom
Multiple R-squared:  0.6521,	Adjusted R-squared:  0.6195 
F-statistic:    20 on 3 and 32 DF,  p-value: 1.741e-07


part i)

y = -36.44 + 0.754 $x_1$ - 1.284 $x_2$ + 1.502 $x_3$    
where:    
$x_1$ -> Whether Software 2 was used  
$x_2$ -> Whether Software 3 was used  
$x_3$ -> Sales from last quarter

Part i)

$E(y | x_1 = 1, x_2 = 0) = (-36.44 + 0.754) - 1.284 * 0 + 1.502 x_3 = -35.69 + 1.502 x_3$   
$E(y | x_1 = 0, x_2 = 1) = (-36.44 - 1.284) + 0.754 * 0 + 1.502 x_3 = -37.72 + 1.502 x_3$ 

In [6]:
anova(sw_model2)

sprintf("Variance explained by the software is %0.3f", (anova(sw_model2)[1,2] / sum(anova(sw_model2)[,2])))

Df Sum Sq    Mean Sq    F value   Pr(>F)      
Software          2  379.5556  189.77778  3.760604 3.412843e-02
SalesLastQuarter  1 2647.8793 2647.87931 52.469921 3.140853e-08
Residuals        32 1614.8707   50.46471        NA           NA

[1] "Variance explained by the software is 0.082"

$H_0$ states that software has no effect on sales.  
$H_A$ states that software does have an effect on sales.

The *f statistic* given from the anova table, 3.76, and the corresponding *p value*, 0.0341, indicates that we do have enough evidence to reject the null hypothesis at the 0.05 significance level.

In [9]:
#5C
sw_model3 <- lm(SalesThisQuarter ~ Software + SalesLastQuarter:Software + SalesLastQuarter, data = df_software)
summary(sw_model3)


Call:
lm(formula = SalesThisQuarter ~ Software + SalesLastQuarter:Software + 
    SalesLastQuarter, data = df_software)

Residuals:
    Min      1Q  Median      3Q     Max 
-9.6222 -3.9426  0.8822  2.8198 11.7895 

Coefficients:
                           Estimate Std. Error t value Pr(>|t|)    
(Intercept)                -92.2593    20.6075  -4.477 0.000102 ***
Software2                  144.6516    31.9049   4.534 8.66e-05 ***
Software3                   48.1097    29.6808   1.621 0.115504    
SalesLastQuarter             2.2122     0.2614   8.462 1.92e-09 ***
Software2:SalesLastQuarter  -1.8579     0.4106  -4.525 8.88e-05 ***
Software3:SalesLastQuarter  -0.6239     0.3879  -1.609 0.118200    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 5.649 on 30 degrees of freedom
Multiple R-squared:  0.7938,	Adjusted R-squared:  0.7594 
F-statistic:  23.1 on 5 and 30 DF,  p-value: 1.849e-09


y = -36.22 + 1.504 $x_1$ + 1.504 $x_2$ + 1.482 $x_3$  
where:    
$x_1$ -> Interaction term between sales last quarter and software 1    
$x_2$ -> Interaction term between sales last quarter and software 2  
$x_3$ -> Interaction term between sales last quarter and software 4  

Part i)

$E(y | x_1 = 1, x_2 = 0, x_3 = 0) = (-36.22 + 1.504) - 7.67 * 0 = -34.72 $   
$E(y | x_1 = 0, x_2 = 1, x_3 = 0) = (-36.22 + 1.504) - 2.00 * 0 = -34.72 $  
$E(y | x_1 = 0, x_2 = 0, x_3 = 1) = (-36.22 + 1.482) - 2.00 * 0 = -34.74$ 

In [8]:
null_model1 <- lm(SalesThisQuarter ~ NULL, data = df_software)
knitr::kable(anova(null_model1, sw_model1))
knitr::kable(anova(sw_model1))

SST <- sum((df_software$SalesThisQuarter - mean(df_software$SalesThisQuarter))^2)
SSR <- sum((sw_model1$fitted.values - mean(df_software$SalesThisQuarter))^2)
SSE <- sum((df_software$SalesThisQuarter - sw_model1$fitted.values)^2)
MSR <- SSR/(sw_model1$rank - 1)
MSE <- SSE/(nrow(df_software) - sw_model1$rank)
df_vector <- c(sw_model1$rank - 1, nrow(df_software) - sw_model1$rank, nrow(df_software) - 1)
SS <- c(SSR, SSE, SST)

anova_table <- as_tibble(cbind(df_vector, SS))

anova_table <- anova_table %>% mutate(MS = SS/df_vector)

anova_table[3,3] <- NA

F_stat <- c(as.numeric(anova_table[1,3] / anova_table[2,3]), NA, NA)

p_value <- c(1 - (pf(F_stat[1], (sw_model1$rank - 1) , (nrow(df_software) - sw_model1$rank))), NA, NA)

anova_table <- cbind(anova_table, F_stat, p_value)

knitr::kable(anova_table)



| Res.Df|      RSS| Df| Sum of Sq|        F|    Pr(>F)|
|------:|--------:|--:|---------:|--------:|---------:|
|     35| 4642.306| NA|        NA|       NA|        NA|
|     33| 4262.750|  2|  379.5556| 1.469161| 0.2447802|



|          | Df|    Sum Sq|  Mean Sq|  F value|    Pr(>F)|
|:---------|--:|---------:|--------:|--------:|---------:|
|Software  |  2|  379.5556| 189.7778| 1.469161| 0.2447802|
|Residuals | 33| 4262.7500| 129.1742|       NA|        NA|



| df_vector|        SS|       MS|   F_stat|   p_value|
|---------:|---------:|--------:|--------:|---------:|
|         2|  379.5556| 189.7778| 1.469161| 0.2447802|
|        33| 4262.7500| 129.1742|       NA|        NA|
|        35| 4642.3056|       NA|       NA|        NA|